packages/components/pie-link/src/index.ts

```diff
@@ -47,29 +49,46 @@ export class PieLink extends LitElement implements LinkProps {
     @property({ type: Boolean })
     public isStandalone = false;
 
+    @property()
+    @validPropertyValues(componentSelector, buttonTypes, 'submit')
+    public type: LinkProps['type'] = 'submit';
+
     render () {
         const {
-            variant, size, iconPlacement, href, target, rel, isBold, isStandalone,
+            tag,
+            variant,
+            size,
+            iconPlacement,
+            isBold,
+            isStandalone,
+            href,
+            target,
+            rel,
+            type,
         } = this;
 
+        const element = unsafeStatic(tag);
+        const isButton = tag === 'button';
+
         return html`
-            <a
+            <${element}  
                 data-test-id="pie-link"
                 class="c-link"
+                tag="${tag}"
                 variant=${variant}
                 size=${size}
-                href=${ifDefined(href)}
-                target=${ifDefined(target)}
-                rel=${ifDefined(rel)}
                 ?isBold=${isBold}
-                ?isStandalone=${isStandalone}>
-                    <div class="c-link-content">
-                      ${iconPlacement === 'leading' ? html`<slot name="icon"></slot>` : nothing}
-                      <slot></slot>
-                      ${iconPlacement === 'trailing' ? html`<slot name="icon"></slot>` : nothing}
-                    </div>
-                </div>
-            </a>`;
+                ?isStandalone=${isStandalone}
+                href=${!isButton && href ? href : nothing}
```

Comment:
Just a thought. Is it worth separating the button and anchor attributes out and then passing them in on a condition instead, it might make better for reading? Something like below (not tested it tho):

```javascript
    const anchorValues = !isButton ? { href, target, rel } : {};
    const buttonValues = isButton ? { type } : {};

    return html`
        <${element} ...${commonAttrs} ...${anchorAttrs} ...${buttonAttrs}>
            `;
......
.....
}
```

Reply:
I have thought about the same thing but unfortunately lit doesn't support spreading attributes yet

https://github.com/lit/lit/issues/923

Reply:
Dam. OK didn't know that.